velox/exec/OutputBuffer.h

```diff
@@ -74,6 +74,36 @@ class ArbitraryBuffer {
   std::deque<std::shared_ptr<SerializedPage>> pages_;
 };
 
+/// The stats of a DestinationBuffer. The data transferred by the buffer has
+/// two phases:
+/// 1. Buffered: after added via enqueue() and before acked, the data is
+///              buffered in the destination buffer.
+/// 2. Sent: the data is sent and removed from the buffer after it is acked
+///          via acknowledge() or deleteResults().
+struct BufferStats {
+  /// Update stats when a data page is added.
+  void recordAddPage(const SerializedPage& data);
+
+  /// Update stats when a data page is sent (removed from buffer).
+  void recordSendPage(const SerializedPage& data);
```

Comment:
recordAcknowledge + recordDelete (the implementation of these two will be the same, but the API will be clearer)

Reply:
done.